### PR TITLE
OCPBUGS-17422: switch to continuously polling NAV-CLOCK

### PR DESCRIPTION
**Pull request by josephdrichard**

### CODE REVIEW

**Comment by josephdrichard** [comment]
/hold
Still need to test

File: addons/intel/e810.go [hunk 1]
```diff
@@ -50,6 +50,14 @@ done
 echo "Disabled all SMA and U.FL Connections"
 `
 
+func getDefaultUblxCmds() []E810UblxCmds {
+	defaultUblxCommands := E810UblxCmds{
+		ReportOutput: false,
+		Args:         []string{"-p", "CFG-MSG,1,34,1"},
```
Comment (jzding) [inline addons/intel/e810.go:56]: add comments:
Set the receiver to output the message NAV-CLOCK once every nav solution.

**Comment by jzding** [comment]
The NAV-STATUS and NAV-CLOCK are pulling at the same cadence, why not also change NAV-STATUS to continuous pulling?

File: pkg/ublox/ublox.go [hunk 2]
```diff
@@ -226,19 +250,70 @@ func match(stdout string, ubLoxRegex *regexp.Regexp) (string, error) {
...
 
-// GetNavOffset ... get gnss offset
 func (u *UBlox) GetNavOffset() (string, error) {
-	args := []string{"-t", "-p", "NAV-CLOCK", "-P", "29.20"}
 
-	output, err := exec.Command(UBXCommand, args...).Output()
-	if err != nil {
-		return "", fmt.Errorf("error executing ubxtool command: %s", err)
-	}
+	output, _ := u.navPoll.UbloxPoll(1000000000, "tAcc")
```
Comment (jzding) [inline pkg/ublox/ublox.go:255]: use `0` to wait forever.

Comment (josephdrichard) [reply pkg/ublox/ublox.go:255]: Support for that (and also for larger numbers) was broken when I tested.  This should wait about 30 years so would be sufficient.

File: pkg/daemon/gpsd.go [hunk 3]
```diff
@@ -52,8 +51,7 @@ type GPSDSubscriber struct {
 
 // Monitor ...
 func (s GPSDSubscriber) Monitor() {
```
[the remark this answers is not in the document]
Comment (josephdrichard) [reply pkg/daemon/gpsd.go:54]: Required to implement  Monitor.  When removed:
*GPSDSubscriber does not implement "github.com/openshift/linuxptp-daemon/pkg/event".Subscriber (missing method Monitor)

**Comment by josephdrichard** [comment]
> The NAV-STATUS and NAV-CLOCK are pulling at the same cadence, why not also change NAV-STATUS to continuous pulling?

done

**Review by jzding — APPROVED** [review]

File: pkg/daemon/gpsd.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/daemon/gpsd.go:54]: we need Monito, monitor is called from event handler

File: pkg/daemon/gpsd.go [hunk 4]
```diff
@@ -72,7 +71,8 @@ func (s GPSDSubscriber) Notify(source event.EventSource, state event.PTPState) {
 // MonitorProcess ... Monitor GPSD process
 func (g *GPSD) MonitorProcess(p config.ProcessConfig) {
 	g.processConfig = p
-	//go g.MonitorGNSSEventsWithGPSD(p, "E810")
+	glog.Info("Starting GNSS Monitoring")
+	go g.MonitorGNSSEventsWithUblox(p, "E810")
```
Comment (aneeshkp) [inline pkg/daemon/gpsd.go:75]: this should move to monitoring

File: pkg/ublox/ublox.go [hunk 5]
```diff
@@ -226,17 +211,51 @@ func match(stdout string, ubLoxRegex *regexp.Regexp) (string, error) {
...
+			u.active = false
+		}
+		go u.UbloxPollPushThread()
 	}
+}
 
-	offset := extractOffset(string(output))
-	return offset, nil
+func (u *UBlox) UbloxPollPushThread() {
+	for {
```
Comment (aneeshkp) [inline pkg/ublox/ublox.go:245]: why two for loop ?

**Review by aneeshkp — APPROVED** [review]
/lgtm

**Pull request merged**